### PR TITLE
Payroll remove id service mock

### LINKED ISSUE #74: Remove ID Service mock
Child of #58

Use the add employee form to add user data manually instead of mocked ID service

### PR DESCRIPTION
**Pull request by sgobotta**
This PR provides:

+ Add employee form refactor
+ Tests for the new form elements

Closes #74

### CODE REVIEW

**Comment by sgobotta** [comment]
Currently fixing lint issues

File: future-apps/payroll/app/src/panels/AddEmployee.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const Form = styled.form`
   grid-template-columns: 1fr 1fr;
   column-gap: 20px;
 
-  > :first-child, > :nth-last-child(-n+2) {
+  > :first-child, > :nth-last-child(-n+1) {
```
Comment (lmcorbalan) [inline future-apps/payroll/app/src/panels/AddEmployee.js]: This is a good chance to move styled components at the bottom of the file

File: future-apps/payroll/app/src/panels/AddEmployee.js [hunk 2]
```diff
@@ -27,93 +27,99 @@ const Form = styled.form`
 
 class AddEmployee extends React.PureComponent {
   static initialState = {
-    entity: null,
-    salary: null,
+    entity: '',
```
Comment (lmcorbalan) [inline future-apps/payroll/app/src/panels/AddEmployee.js]: What about naming this as `address`?

**Comment by sgobotta** [comment]
I've been working on removing the mocked data and a few things came to my mind.

At the moment, when we create an employee, the event coming from the contract returns an employee with some properties, including 'name', which is not the Full Name we currently use in the updated form, but the Domain Name (name.organization.eth).

We should decide what the form behaviour will be. Do we want to fill in a domain name or the full name of an employee? Should we validate the address given in the form exists in a mocked data file or should we accept any address? (except those that are already used)

**Comment by lmcorbalan** [comment]
> At the moment, when we create an employee, the event coming from the contract returns an employee with some properties, including 'name', which is not the Full Name we currently use in the updated form, but the Domain Name (name.organization.eth).

It was because we were sending to the contract the ENS in place of the full name, but now as we send the full name we'll receive it in the event

> We should decide what the form behaviour will be. Do we want to fill in a domain name or the full name of an employee? Should we validate the address given in the form exists in a mocked data file or should we accept any address? (except those that are already used)

Regarding address validation, I think we should checks if the given string is an address (https://github.com/aragon/aragon-ui/blob/master/src/utils/web3.js#L56) and it's not being used by an existing employee

**Comment by sgobotta** [comment]
> > At the moment, when we create an employee, the event coming from the contract returns an employee with some properties, including 'name', which is not the Full Name we currently use in the updated form, but the Domain Name (name.organization.eth).
> 
> It was because we were sending to the contract the ENS in place of the full name, but now as we send the full name we'll receive it in the event

Alright. Then we should also change the mocked data in the PayrollKit to create accounts with name. Are we still going to use the ENS? If so, we should have the proper functions in the contract to create employees with name and ENS.

> > We should decide what the form behaviour will be. Do we want to fill in a domain name or the full name of an employee? Should we validate the address given in the form exists in a mocked data file or should we accept any address? (except those that are already used)
> 
> Regarding address validation, I think we should checks if the given string is an address (https://github.com/aragon/aragon-ui/blob/master/src/utils/web3.js#L56) and it's not being used by an existing employee

Understood, that is already done.

**Pull request merged**